src/aind_metadata_service/sharepoint/nsb2023/mapping.py

```diff
@@ -1479,12 +1906,114 @@ def aind_inj4ret_setting(self) -> Optional[Any]:
+        """Maps inj5_ionto_time to aind model"""
+        return self._parse_length_of_time_str(self._nsb.inj5_ionto_time)
+
+    @property
+    def aind_inj5_storage_location(self) -> Optional[str]:
+        """Maps inj5_storage_location to aind model"""
+        return self._nsb.inj5_storage_location
+
+    @property
+    def aind_inj5_type(self) -> Optional[InjectionType]:
+        """Maps inj5_type to aind model"""
+        return (
+            None
+            if self._nsb.inj5_type is None
+            else {
+                self._nsb.inj5_type.SELECT: None,
+                self._nsb.inj5_type.IONTOPHORESIS: InjectionType.IONTOPHORESIS,
+                self._nsb.inj5_type.NANOJECT_PRESSURE: InjectionType.NANOJECT,
+            }.get(self._nsb.inj5_type, None)
+        )
+
+    @property
+    def aind_inj5_virus_strain_rt(self) -> Optional[str]:
+        """Maps inj5_virus_strain_rt to aind model"""
+        return self._nsb.inj5_virus_strain_rt
+
+    @property
+    def aind_inj5ret_setting(self) -> Optional[Any]:
+        """Maps inj5ret_setting to aind model"""
+        return (
+            None
+            if self._nsb.inj5ret_setting is None
+            else {
+                self._nsb.inj5ret_setting.OFF: None,
+                self._nsb.inj5ret_setting.ON: None,
+            }.get(self._nsb.inj5ret_setting, None)
+        )
+
+    @property
+    def aind_inj5volperdepth(self) -> Optional[float]:
+        """Maps inj5volperdepth to aind model"""
+        return self._nsb.inj5volperdepth
+
+    @property
+    def aind_inj6_alternating_time(self) -> Optional[float]:
+        """Maps inj6_alternating_time to aind model"""
+        return self._parse_alt_time_str(self._nsb.inj6_alternating_time)
+
+    @property
+    def aind_inj6_current(self) -> Optional[float]:
+        """Maps inj6_current to aind model"""
+        return self._parse_current_str(self._nsb.inj6_current)
+
+    @property
+    def aind_inj6_ionto_time(self) -> Optional[float]:
+        """Maps inj6_ionto_time to aind model"""
+        return self._parse_length_of_time_str(self._nsb.inj6_ionto_time)
+
+    @property
+    def aind_inj6_storage_location(self) -> Optional[str]:
+        """Maps inj6_storage_location to aind model"""
+        return self._nsb.inj6_storage_location
+
+    @property
+    def aind_inj6_type(self) -> Optional[InjectionType]:
+        """Maps inj6_type to aind model"""
+        return {
+            self._nsb.inj6_type.SELECT: None,
+            self._nsb.inj6_type.IONTOPHORESIS: InjectionType.IONTOPHORESIS,
+            self._nsb.inj6_type.NANOJECT_PRESSURE: InjectionType.NANOJECT,
+        }.get(self._nsb.inj6_type, None)
+
+    @property
+    def aind_inj6_virus_strain_rt(self) -> Optional[str]:
+        """Maps inj6_virus_strain_rt to aind model"""
+        return self._nsb.inj6_virus_strain_rt
+
+    @property
+    def aind_inj6ret_setting(self) -> Optional[Any]:
+        """Maps inj6ret_setting to aind model"""
+        return {
```

Comment:
You'll need the return None if here

Reply:
agh, I thought I caught them all this time. Thanks for finding these

Reply:
i went through the mapping and (hopefully) caught everything this time.